### PR TITLE
alert missing rmagick on method call, rather than on load

### PR DESCRIPTION
**Pull request by jordanstephens**
Following @nadavshatz advice on improving the rmagick dependency warning.

This actually solves #9 even though I closed that issue early.

### CODE REVIEW

File: lib/paleta.rb [hunk 1]
```diff
@@ -3,4 +3,13 @@
 require 'paleta/palette'
 
 module Paleta
+  @rmagick_available = begin
```
Comment (nadavshatz) [inline lib/paleta.rb]: shouldn't this be a class variable ? `@@rmagick_available`

File: lib/paleta/palette.rb [hunk 2]
```diff
@@ -237,6 +209,25 @@ def fit
 
     private
 
+    def self.generate_from_image(path, size = 5)
+      unless Paleta.rmagick_available?
+        return puts "You must install RMagick to use Palette.generate(:from => :image, ...)"
```
Comment (nadavshatz) [inline lib/paleta/palette.rb]: I'm thinking this should probably `raise` / `fail` as the method is completely unusable.

**Comment by nadavshatz** [comment]
@jordanstephens Look good and a lot cleaner than what we have today. I wrote a couple of small comments.
We should probably add tests for this to the suit as well.

P.S
I wanted to add code-climate/hound-ci and circle-ci/travis-ci but i don't have the right privileges for the repo. Another thing that could be useful is gemnasium for automatic detection of stale dependencies.

File: lib/paleta.rb (hunk 1 above)
Comment (jordanstephens) [inline lib/paleta.rb]: I'm not sure—I generally avoid using class variables. If I remember correctly, the differences between the two show up when inheritance is at play. That shouldn't apply here, but I may need a refresher on when it is best to use each one. Is there a specific case you're concerned about?

File: lib/paleta/palette.rb (hunk 2 above)
Comment (jordanstephens) [inline lib/paleta/palette.rb]: Yep, I agree. I'll change it to raise an exception.

**Comment by jordanstephens** [comment]
I went ahead and pushed an update to raise the exception, and added a spec with a stubbed `Paleta.rmagick_available?`. I'm not totally sure if there is a better way to test this or not. If you have any better ideas, I'm totally game.

I totally support adding those things, let me see if I can set them up. Is there something I can change to give you the correct rights? I have you added as a collaborator, but it doesn't look like I have any other options.

**Comment by jordanstephens** [comment]
@nadavshatz just added travis, code climate, and gemnasium :+1: 

code climate is a little sad though :disappointed:

**Comment by nadavshatz** [comment]
Found this regarding the class variables http://makandracards.com/makandra/14229-the-many-gotchas-of-ruby-class-variables
Looks like you are right, I stand corrected :smiley: 
This all looks great to me, awesome job :+1: 

It's great that we have these things now, and we can work to slowly improve on code climate, at least it gives us a point of reference.

**Comment by jordanstephens** [comment]
Cool, thanks @nadavshatz!

**Comment by jordanstephens** [comment]
Just pushed v0.3.0 with these changes. Decided to bump the minor version rather than the tiny version because of the change to raise when missing rmagick on `generate_from_image`.

**Pull request merged**